datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -4630,4 +4645,125 @@ digraph {
         let parameter_type = params.clone().get(placeholder_value).unwrap().clone();
         assert_eq!(parameter_type, None);
     }
+
+    #[test]
+    fn test_join_with_new_exprs() {
```

Comment:
```suggestion
    fn test_join_with_new_exprs() -> Result<()> {
```
Make the function fallible so that many `unwrap`'s can be replaced with `?`